src/StmtToViz.cpp

```diff
@@ -0,0 +1,2900 @@
+            << tag << " " << context_stack.empty() << " " << context_stack_tags.back();
+        context_stack.pop_back();
+        context_stack_tags.pop_back();
+        stream << "</" + tag + ">";
+    }
+
+    // Prints an html element: opening tag, body and closing tag
+    void print_html_element(const string &tag, const string &cls, const string &body, int id = -1) {
+        print_opening_tag(tag, cls, id);
+        stream << body;
+        print_closing_tag(tag);
+    }
+
+    void print_html_element(const string &tag, const string &cls, const string &body, string id) {
+        print_opening_tag(tag, cls, id);
+        stream << body;
+        print_closing_tag(tag);
+    }
+
+    // Prints the opening/closing tags for an anchor that toggles code block view
+    void print_toggle_anchor_opening_tag(int id) {
+        stream << "<a onclick='return toggle(" << id << ");' href=_blank>";
+    }
+
+    void print_toggle_anchor_closing_tag() {
+        stream << "</a>";
+    }
+
+    // Prints a variable to stream
+    void print_variable(const string &x) {
+        stream << variable(x);
+    }
+
+    string variable(const string &x) {
+        int id;
+        if (scope.contains(x)) {
+            id = scope.get(x);
+        } else {
+            id = gen_unique_id();
+            scope.push(x, id);
+        }
+        ostringstream s;
+        s << "<b class='variable matched' id='" << id << "-" << gen_unique_id() << "'>";
+        s << x;
+        s << "</b>";
+        return s.str();
+    }
+
+    // Prints text to stream
+    void print_text(const string &x) {
+        stream << x;
+    }
+
+    // Prints the button to show or hide a code scope
+    void print_show_hide_icon(int id) {
+        stream << "<div class='show-hide-btn-wrapper'>"
+               << "  <div class='show-hide-btn' style='display:none;' id=" << id << "-show>"
+               << "    <i class='bi bi-plus-square' title='Expand code block'></i>"
+               << "  </div>"
+               << "  <div class='show-hide-btn' id=" << id << "-hide>"
+               << "    <i class='bi bi-dash-square' title='Collapse code block'></i>"
+               << "  </div>"
+               << "</div>";
+    }
+
+    // Prints a button to sync text with visualization
+    void print_visualization_button(string id) {
+        stream << "<button class='icon-btn sync-btn' onclick='scrollToViz(\"" << id << "\")'>"
+               << "  <i class='bi bi-arrow-right-square' title='Jump to visualization'></i>"
+               << "</button>";
+    }
+
+    // Maaz: This is a legacy function from the old html generator. I made sure it works
+    // with the refactored code but I did not dig into why this function does what it does.
+    void print_cuda_gpu_source_kernels(const string &str) {
+        print_opening_tag("code", "ptx");
+
+        int current_id = -1;
+        bool in_braces = false;
+        bool in_func_signature = false;
+
+        string current_kernel;
```

Comment:
style nit: IMHO, within a given source file, we should prefer to either prefix all the `std` stuff explicitly, or put `using` at the top of the file and omit the prefixes ~everywhere. Mixing the two in this way isn't wrong but makes the code harder to read -- choose one style and stick with it for the file.